packages/@lwc/ssr-compiler/src/compile-template/component.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { builders as b, is } from 'estree-toolkit';
+import { kebabcaseToCamelcase, toPropertyName } from '@lwc/template-compiler';
+import { esTemplateWithYield } from '../estemplate';
+import { cleanStyleAttrVal, isValidIdentifier } from './shared';
+import { TransformerContext } from './types';
+import { expressionIrToEs } from './expression';
+
+import type {
+    BlockStatement as EsBlockStatement,
+    ObjectExpression as EsObjectExpression,
+} from 'estree';
+import type {
+    Attribute as IrAttribute,
+    Component as IrComponent,
+    Property as IrProperty,
+} from '@lwc/template-compiler';
+import type { Transformer } from './types';
+
+const bYieldFromChildGenerator = esTemplateWithYield<EsBlockStatement>`
+    {
+        const childProps = ${is.objectExpression};
+        const childAttrs = ${is.objectExpression};
+        const childSlottedContentGenerators = {};
+        yield* ${is.identifier}(${is.literal}, childProps, childAttrs, childSlottedContentGenerators);
+    }
+`;
+
+const bImportGenerateMarkup = (localName: string, importPath: string) =>
+    b.importDeclaration(
+        [b.importSpecifier(b.identifier('generateMarkup'), b.identifier(localName))],
+        b.literal(importPath)
+    );
+
+function getChildAttrsOrProps(
+    attrs: (IrAttribute | IrProperty)[],
+    cxt: TransformerContext
+): EsObjectExpression {
+    const objectAttrsOrProps = attrs.map((attr) => {
+        const key = isValidIdentifier(attr.name) ? b.identifier(attr.name) : b.literal(attr.name);
+        if (attr.value.type === 'Literal' && typeof attr.value.value === 'string') {
+            const value =
+                attr.name === 'style' ? cleanStyleAttrVal(attr.value.value) : attr.value.value;
+            return b.property('init', key, b.literal(value));
+        } else if (attr.value.type === 'Literal' && typeof attr.value.value === 'boolean') {
+            return b.property('init', key, b.literal(attr.value.value));
+        } else if (attr.value.type === 'Identifier' || attr.value.type === 'MemberExpression') {
+            const propValue = expressionIrToEs(attr.value, cxt);
+            return b.property('init', key, propValue);
+        }
```

Comment:
These are IR nodes rather than estree AST nodes. We don't currently have any helpers like `is.literal` that operate on those types. When [template expressions](https://github.com/salesforce/lwc/pull/3376) goes GA, these will be true JS expressions; at that point, we should be able to use off-the-shelf util functions like `is.literal`.